src/gccrs/mod.rs

```diff
@@ -73,4 +77,40 @@ impl Gccrs {
         Gccrs::dump_config()?;
         GccrsConfig::display()
     }
+
+    fn spawn_with_args(args: &Vec<String>) -> Result<ExitStatus> {
+        Command::new("gccrs").args(args).status()
+    }
+
+    fn compile(args: &Vec<String>) -> Result {
+        let mut gccrs_args = GccrsArgs::from_rustc_args(args);
+
+        gccrs_args
+            .drain(..)
+            .map(|arg_set| Gccrs::spawn_with_args(&arg_set.to_args()))
+            .map(|exit_status| match exit_status?.success() {
+                true => Ok(()),
+                // FIXME: Display given arguments
+                false => Err(Error::new(
+                    ErrorKind::Other,
+                    "Couldn't compile program with given arguments",
+                )),
+            })
+            .collect::<Result>()
+    }
+
+    /// Convert arguments given to `rustc` into valid arguments for `gccrs`
+    pub fn handle_rust_args(args: &Vec<String>) -> Result {
+        let first_rustc_arg = args.get(2);
+
+        match first_rustc_arg.map(|s| s.as_str()) {
+            // FIXME: Is that true? Should we use getopts and just parse it and convert
+            // it as well?
+            // If rustc is invoked with stdin as input, then it's simply to print the
+            // configuration option in our case, since we are compiling a rust project
+            // with files and crates
+            Some("-") => Gccrs::cfg_print(),
```

Comment:
- indicates read from stdin. When you use `--print` it is just necessary to provide an input file.

Reply:
My question regarding the `FIXME` was related to cargo's usage of the "-" argument: Am I right in assuming that it will only ask rustc to read from `stdin` when initially printing the configuration? There is no weird, hidden flag to compile from stdin that I'm unaware about?

Reply:
I don't think cargo will ever ask rustc to read from stdin except for when using `--print` or maybe if using something weird like `[lib] path = "-"` in `Cargo.toml`. It os still cleaner to check for the right `--print` instead though.

Reply:
I tried it in an empty project, and `cargo` doesn't build anything with `[lib] path = "-"`. It's invoking rustc/gccrs just like during compilation, which is without redirecting stdin to the spawned process's stdin.

I agree with you that it would be cleaner, but I think it should wait for now. Doing it this way is really practical since we can generate the `gccrs.target_options.dump` and parse it. Using the `GccrsArgs` struct and converting `--print=cfg` into `-frust-dump-target_options` would be nice, but it would require some sort of callback or other means to implement the parsing and printing of the configuration. Or, as discussed with @philberty, we could also make it so that gccrs displays target options in the same way as rustc, and on stdout instead of in a file. I've opened #7 to remember to get to this at some point